apps/common/controllers/add-another-address-loop.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const _ = require('lodash');
+const BaseController = require('./base');
+
+module.exports = class AddAnotherAddressController extends BaseController {
+  locals(req, res, callback) {
+    const locals = super.locals(req, res, callback);
+    const field = locals.field;
+    const addresses = req.sessionModel.get(`${field}Addresses`);
+    const hasAddresses = req.sessionModel.get(`${field}Addresses`) ? true : false;
+    const items = [];
+    _.forEach(addresses, (value, key) => {
+      const address = {
+        id: key,
+        address: value.address
+      };
+      items.push(address);
+    });
+    return Object.assign({}, locals, {
+      items,
+      hasAddresses
+    });
+  }
+
+  get(req, res, callback) {
+    if (req.params.action === 'delete' && req.params.id) {
+      const field = this.options.locals.field;
```

Comment:
if we are relying on field being set throughout this controller we should probably set a default in the constructor, or throw an error